### PR TITLE
Support isRunningInAppCenterTest()

### PR DESCRIPTION
**Pull request by owenniblock**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Creates feature parity in the `android` and `apple` SDKs so that the `isRunningInAppCenterTestCloud()` can be surfaced in the Xamarin SDK.

## Related PRs or issues

- [Original Android SDK PR is here](https://github.com/microsoft/appcenter-sdk-android/pull/1171)

## Misc

It seemed overkill to add tests for something which is essentially doing a very simple reference to Apple's environment variables. Please let me know if you need me to do this - it will require:
- Adding something to Sasquatch app to show whether `isRunningInAppCenterTestCloud()` returns true or false.
- A test to verify the value in the UI.

### CODE REVIEW

**Comment by owenniblock** [comment]
I'm getting some test failures. I don't think they're related to my changes but I'll check them out when I get a mo.

**Comment by clpolet** [comment]
CI failure was caused by some signing issues when the PR is from a fork. Should work now. Closing/reopening.

**Comment by owenniblock** [comment]
Cool, do you need me to do anything else to get this in? Once it's there I'll add the bindings for the Xamarin SDK project.

File: AppCenter/AppCenter/Internals/MSAppCenterInternal.h [hunk 1]
```diff
@@ -24,6 +24,16 @@ static NSString *const kMSDisableVariable = @"APP_CENTER_DISABLE";
...
 
+/**
+ * Name of the environment variable to check to see if we're running in App Center Test.
+ */
+static NSString *const kMSRunningInAppCenter = @"RUNNING_IN_APP_CENTER";
+
+/**
+ * A string value for environment variables denoting `true`.
+ */
+static NSString *const kMSTrue_Environment_String = @"1";
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/MSAppCenterInternal.h]: Wrong style for the constant name.

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,12 @@ ___
 
 ___
 
+## Version 2.2.0 (Under development)
+
+### App Center
+
+* **[Feature]** Add `AppCenter.isRunningInAppCenterTestCloud` to provide method to check if the application is running in Test Cloud.
```
Comment (jaelim-ms) [inline CHANGELOG.md]:
```suggestion
* **[Feature]** Add `isRunningInAppCenterTestCloud` in `MSAppCenter` to provide method to check if the application is running in Test Cloud.
```

File: AppCenter/AppCenter/MSAppCenter.h [hunk 3]
```diff
@@ -77,6 +77,13 @@
  */
 + (BOOL)isConfigured;
 
+/**
+ * Check whether app is running in App Center Test.
+ *
+ * @return true if running in App Center Test, false otherwise.
+ */
++ (BOOL)isRunningInAppCenterTestCloud;
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/MSAppCenter.h]: The doc says App Center Test but the method name says Test Cloud. Need it to be consistent.

File: AppCenter/AppCenter/MSAppCenter.m [hunk 4]
```diff
@@ -106,6 +106,15 @@ + (BOOL)isConfigured {
   return [MSAppCenter sharedInstance].sdkConfigured && [MSAppCenter sharedInstance].configuredFromApplication;
 }
 
++ (BOOL)isRunningInAppCenterTestCloud {
+    NSDictionary *environmentVariables = [[NSProcessInfo processInfo] environment];
+    NSString *runningInAppCenter = environmentVariables[kMSRunningInAppCenter];
+    if (runningInAppCenter == kMSTrue_Environment_String) {
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/MSAppCenter.m]: String comparison should be done by calling `isEqualToString:`.
Here is a [doc](https://developer.apple.com/documentation/foundation/nsstring/1407803-isequaltostring?language=objc).

File: CHANGELOG.md (hunk 2 above)
Comment (jaelim-ms) [reply CHANGELOG.md]: There is a bad merge in the file. It always happens and there is one more in the file. Would you please use one `## Version 2.2.0 (Under development)`?

File: CHANGELOG.md [hunk 5]
```diff
@@ -25,6 +25,12 @@ ___
 
 ___
 
+## Version 2.2.0 (Under development)
+
+### App Center
```
Comment (jaelim-ms) [inline CHANGELOG.md]:
```suggestion
### AppCenter
```

**Comment by owenniblock** [comment]
@jaelim-ms I think this is ready for a re-review 😄

File: AppCenter/AppCenter/MSAppCenter.m [hunk 6]
```diff
@@ -106,6 +106,15 @@ + (BOOL)isConfigured {
   return [MSAppCenter sharedInstance].sdkConfigured && [MSAppCenter sharedInstance].configuredFromApplication;
 }
 
++ (BOOL)isRunningInAppCenterTestCloud {
```
Comment (clpolet) [inline AppCenter/AppCenter/MSAppCenter.m]: I think it's worth adding a small unit test for this. There is a similar unit test [here](https://github.com/microsoft/appcenter-sdk-apple/blob/27b6cdf2c9576ae3655b63488abaa6e9f511f476/AppCenter/AppCenterTests/MSAppCenterTests.m#L455).

Comment (owenniblock) [reply AppCenter/AppCenter/MSAppCenter.m]: Good call @clpolet - test added!

File: CHANGELOG.md [hunk 7]
```diff
@@ -27,6 +27,12 @@
 
 ___
 
+## Version 2.2.0 (Under development)
```
Comment (jaelim-ms) [inline CHANGELOG.md]: Seems like a bad merge.

**Pull request merged**